homeassistant/components/melnor/sensor.py

```diff
@@ -125,9 +123,14 @@ def icon(self) -> str | None:
         if self.entity_description.icon_fn is not None:
             return self.entity_description.icon_fn(self._device)
 
-        return self._attr_icon
+        return super().icon
```

Comment:
Just returning `None` here is clearer IMO

Reply:
Sure!